### PR TITLE
Github

### PR DESCRIPTION
**Pull request by Skullbock**
Added Github Sync. See #57 

schermata 2014-11-08 alle 18 22 22

### CODE REVIEW

File: component/admin/models/languages.php [hunk 1]
```diff
@@ -295,4 +296,46 @@ public function purge()
...
+
+		try 
+		{
+			$contents = $github->repositories->contents->get('joomla', 'joomla-cms', 'language/en-GB', 'staging');
+			$contents = array_merge($contents, $github->repositories->contents->get('joomla', 'joomla-cms', 'administrator/language/en-GB', 'staging'));
+			$contents = array_merge($contents, $github->repositories->contents->get('joomla', 'joomla-cms', 'installation/language', 'staging'));
+		} 
+		catch (Exception $e)
+		{
+			JFactory::getApplication()->enqueueMessage($e);
```
Comment (zero-24) [inline component/admin/models/languages.php]: @Skullbock can you have a look here?

I think we should use here:

``` php
JFactory::getApplication()->enqueueMessage($e->getMessage());
```

see here: http://php.net/manual/en/language.exceptions.php

**Comment by zero-24** [comment]
@Skullbock can you have a look into the CS issues mentioned by Travis:

```
Initializing PHP_CodeSniffer checks.
...........EEEEE.....................WW......
FILE: ...ild/joomla-projects/com_localise/component/admin/models/translation.php
--------------------------------------------------------------------------------
FOUND 1 ERROR(S) AFFECTING 1 LINE(S)
--------------------------------------------------------------------------------
 196 | ERROR | Functions and classes must not contain multiple empty lines in a
     |       | row; found 2 empty lines
--------------------------------------------------------------------------------
FILE: ...ild/joomla-projects/com_localise/component/admin/models/packagefile.php
--------------------------------------------------------------------------------
FOUND 1 ERROR(S) AFFECTING 1 LINE(S)
--------------------------------------------------------------------------------
 146 | ERROR | Doc comment for "$pk" missing
--------------------------------------------------------------------------------
FILE: ...s/build/joomla-projects/com_localise/component/admin/models/package.php
--------------------------------------------------------------------------------
FOUND 1 ERROR(S) AFFECTING 1 LINE(S)
--------------------------------------------------------------------------------
 146 | ERROR | Doc comment for "$pk" missing
--------------------------------------------------------------------------------
FILE: ...build/joomla-projects/com_localise/component/admin/models/languages.php
--------------------------------------------------------------------------------
FOUND 8 ERROR(S) AFFECTING 8 LINE(S)
--------------------------------------------------------------------------------
 309 | ERROR | Instanciating new classes without parameters does not require
     |       | brackets.
 312 | ERROR | Please put a space between the // and the start of comment text;
     |       | found "//$options->set('gh.token', '');"
 313 | ERROR | Whitespace found at end of line
 316 | ERROR | Whitespace found at end of line
 321 | ERROR | Whitespace found at end of line
 325 | ERROR | Please consider an empty line before the return statement;
 327 | ERROR | Whitespace found at end of line
 328 | ERROR | Whitespace found at end of line
--------------------------------------------------------------------------------
FILE: .../build/joomla-projects/com_localise/component/admin/models/language.php
--------------------------------------------------------------------------------
FOUND 2 ERROR(S) AFFECTING 2 LINE(S)
--------------------------------------------------------------------------------
 125 | ERROR | Doc comment for "$pk" missing
 372 | ERROR | Doc comment for "$pks" missing
--------------------------------------------------------------------------------
FILE: .../joomla-projects/com_localise/component/admin/controllers/languages.php
--------------------------------------------------------------------------------
FOUND 0 ERROR(S) AND 1 WARNING(S) AFFECTING 1 LINE(S)
--------------------------------------------------------------------------------
 66 | WARNING | Possible useless method overriding detected
--------------------------------------------------------------------------------
FILE: ...omla-projects/com_localise/component/admin/controllers/translations.php
--------------------------------------------------------------------------------
FOUND 0 ERROR(S) AND 1 WARNING(S) AFFECTING 1 LINE(S)
--------------------------------------------------------------------------------
 38 | WARNING | Possible useless method overriding detected
--------------------------------------------------------------------------------
```

Also some checks fail.
https://travis-ci.org/joomla-projects/com_localise/jobs/40430628

**Comment by infograf768** [comment]
Also look at https://github.com/joomla-projects/com_localise/pull/242
I have some issues with it

**Comment by Skullbock** [comment]
Should've fixed the issues mentioned

**Comment by javigomez** [comment]
Thanks Daniele

Just mention that this is a proof of concept, if anyone has an idea of where to better place the button to fetch the language or a better name for it just say it. I'm aware that @Skullbock has in mind also to allow in the future to fetch specific tags of the different Joomla versions. That will be a great improvement, but for now we would like to concentrate on get this specific feature working and do more progress once is working.

**Comment by infograf768** [comment]
Could you sync your branch as we did a few commits

**Comment by infograf768** [comment]
@Skullbock 
also, can you solve this: https://github.com/joomla-projects/com_localise/pull/242

Note: I tested the part concerning getting en-GB from github and I get:

> API rate limit exceeded for 2.2.50.217. (But here's the good news: Authenticated requests get a higher rate limit. Check out the documentation for more details.)

**Comment by Skullbock** [comment]
@infograf768 i should've merged and resolved conflicts. Also, for github's api limitation, there is a todo there to allow adding the gh.token for using the higher rate limit instead of the free one.

**Comment by infograf768** [comment]
`( ! ) Notice: Undefined property: stdClass::$joomlacopyright in /Applications/MAMP/htdocs/trunkgitnew/administrator/components/com_localise/models/language.php on line 111`

In fact Joomla!™ Copyright should always be present as these are the language xml files, not used by third party

therefore all works fine (for languages) when you take off the conditional
`if (!$data->joomlacopyright) {`

**Comment by infograf768** [comment]
oops posted in wrong PR

**Comment by infograf768** [comment]
Obsolete

**Pull request closed without merging**